### ISSUE TITLE
WithEvents fields are treated as non-fields in naming enforcement

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#15343]
We have a rule that says that Non-field members should be pascal cased, this is firing on all the WithEvents VB fields:

``` VB
Private WithEvents _buildEvents As EnvDTE.BuildEvents
```

**Comment by Pilchie** [comment]
I expect they are treated as properties instead :-/

**Comment by davkean** [comment]
I really think you need to reconsider this - this basically makes WinForms + VB usable with the new naming rules.

**Comment by Pilchie** [comment]
Sorry - meant to include the fact that we think the fix for https://github.com/dotnet/roslyn/issues/15326 is likely to address this for the `_` case, since we won't treat `_` as a lowercase anymore.

@dpoeschl I suppose this would still be a problem if you used VB and *didn't* use underscores, because the upper-case rule would be applied, even though people think of them as fields?

**Comment by davkean** [comment]
Yes this is a problem if you remove the underscore - it's still marked as a naming violation.

**Comment by davkean** [comment]
This looks fixed in Version 16.0.0 Preview 3.0 [28517.116.d16.0stg].

**Comment by davkean** [comment]
Note, my mistake this is still occuring.
